### PR TITLE
Enable nullability in ButtonBase

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in ButtonBase.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6545)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/ButtonBase.cs [hunk 1]
```diff
@@ -1164,7 +1158,7 @@ protected override void OnPaint(PaintEventArgs pevent)
             if (GetStyle(ControlStyles.UserPaint))
             {
                 Animate();
-                ImageAnimator.UpdateFrames(Image);
+                ImageAnimator.UpdateFrames(Image!);
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/ButtonBase.cs:1161]: This at least deserves a comment that [`ImageAnimator`](https://github.com/dotnet/runtime/blob/bc6d349ecd72eec162f0532e6b6218fdbaa07ddb/src/libraries/System.Drawing.Common/src/System/Drawing/ImageAnimator.cs#L103-L108) isn't NRTA'ed, and accepts nulls.

So maybe we should just add a null check and avoid the bang operator at all.

File: src/System.Windows.Forms/src/System/Windows/Forms/ButtonBase.cs [hunk 2]
```diff
@@ -470,7 +471,7 @@ public string ImageKey
                 }
 
                 // If they were previously using indexes - this should clear out the image index field.
-                _imageIndex.Key = value;
+                _imageIndex.Key = value!;
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/ButtonBase.cs:474]: `Indexer` type isn't annotated correctly, as `Key` accepts nulls:
https://github.com/dotnet/winforms/blob/6365af0d23685c467a74bbfeb9363d3ddec3d50c/src/System.Windows.Forms/src/System/Windows/Forms/ImageList.Indexer.cs#L29-L35

**Review by RussKie — CHANGES_REQUESTED** [review]

File: src/System.Windows.Forms/src/System/Windows/Forms/ButtonBase.cs (hunk 1 above)
Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/ButtonBase.cs:1161]: I see in https://github.com/dotnet/runtime/blob/7d30da68a91d1cced11a0302cc7a8aa9a0ea9af1/src/libraries/System.Drawing.Common/src/System/Drawing/ImageAnimator.cs#L103 there is already a null check and that nullability is enabled in https://github.com/dotnet/runtime/blob/7d30da68a91d1cced11a0302cc7a8aa9a0ea9af1/src/libraries/System.Drawing.Common/src/System/Drawing/ImageAnimator.cs#L103.

Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/ButtonBase.cs:1161]: The actual type doesn't look like it was annotated (correctly). @jeffhandley would you know the story behind it?

Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/ButtonBase.cs:1161]: Perhaps I should just add the null check on our side to unblock this PR?

Comment (jeffhandley) [reply src/System.Windows.Forms/src/System/Windows/Forms/ButtonBase.cs:1161]: Sorry; I'm overlooking the issue. It looks to me like ImageAnimator.UpdateFrames expects a non-null Image, but it gracefully no-ops if null is passed in; that seems OK to me--is there an issue with that approach?

My recommendation would be to add the null check on this call site.

**Review by dreddy-work — APPROVED** [review]
LGTM.

File: src/System.Windows.Forms/src/System/Windows/Forms/ButtonBase.cs (hunk 1 above)
Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/ButtonBase.cs]: @jeffhandley my understanding of NRTA is that if a method accepts nulls (like in this situation) it should be annotated as nullable.

Comment (jeffhandley) [reply src/System.Windows.Forms/src/System/Windows/Forms/ButtonBase.cs]: > @jeffhandley my understanding of NRTA is that if a method accepts nulls (like in this situation) it should be annotated as nullable.

Sorry for the delayed response, @RussKie. OK, I'm convinced. :wink:. I'll submit a PR to change it to `public static void ImageAnimator.UpdateFrames(Image? image)`.

Comment (jeffhandley) [reply src/System.Windows.Forms/src/System/Windows/Forms/ButtonBase.cs]: Created https://github.com/dotnet/runtime/pull/65949 to change the annotation.

Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/ButtonBase.cs]: Thank you!

**Pull request merged**